김현정/week16/back_tracking/쿼드압축 후 개수 세기.java

```diff
@@ -0,0 +1,56 @@
+package back_tracking;
+
+/**
+ * Author    : 김현정
+ * Date      : 2024.09.05(목)
+ * Runtime   :
+ * Memory    :
+ * Algorithm : [programmers]백트래킹 - LV2_68936 쿼드 압축 후 세기
+ */
+class Solution {
+    public static void main(String[] args) {
+        Solution solution = new Solution();
+        int[] answer = solution.solution(new int[][]{{1,1,0,0},{1,0,0,0},{1,0,0,1},{1,1,1,1}});
+        for(int a : answer){
+            System.out.println(a);
+        }
+    }
+
+    static int[] answer;
+    public int[] solution(int[][] arr) {
+        //전체 압축이 가능한지 볼 것
+        //안되면 반으로 줄일 것
+        answer = new int[2];
+        dfs(arr, 0, 0, arr.length);
+
+        return answer;
+    }
+
+    public void dfs(int[][] arr, int x, int y, int size) {
+        boolean flag = true;
+
+        for(int i = x; i < x+ size; i++) {
+            for(int j = y; j < y + size; j++) {
+                if(arr[i][j] != arr[x][y]) {
+                    flag = false;
+                    break;
+                }
+            }
+            if(!flag) break;
+        }
```

Comment:
```suggestion
        boolean flag = true;

        NESTED_LOOP:
        for(int i = x; i < x+ size; i++) {
            for(int j = y; j < y + size; j++) {
                if(arr[i][j] != arr[x][y]) {
                    flag = false;
                    break NESTED_LOOP;
                }
            }
        }
```

리뷰 하면서 코드 보다가 생각나서 저도 수정 했는데, 2중 루프를 탈출을 다른 방법으로 하는 코드로 제 코드를 수정 하면서 리뷰로 남깁니다.

어떤 방식이던 큰 차이는 없어 보이네요.

Reply:
오 이렇게도 풀 수 있군요! 하나 알아갑니다! 감사합니다! 👍